tests/core/pthread/test_pthread_dylink_tls.out

```diff
@@ -0,0 +1,13 @@
+main
+side_tls value  : 42
+main_tls value  : 10
+increment done
+side_tls value  : 62
+main_tls value  : 30
+thread_main
+side_tls value  : 42
+main_tls value  : 10
+increment done
+side_tls value  : 52
+main_tls value  : 20
+success
```

Comment:
it's probably my own lack of knowledge here - is it valid to read the side module's TLS before starting a thread? (before `thread_main` is printed here) What does it actually get when reading TLS in that manner - is that "another" TLS on the main thread in some sense?

Reply:
Yes, the main module and side module TLS work find with just one thread...     The main thread TLS should be completely unrelated to the second thread and unaffected by the running of code on the second thread.  That is kind of point of this test.

I'll update the output to be a little more self-explanatory.

Reply:
I think the terminology might be a little confusing here because we have "main thread" and "side thread" and they both run the "main module" and "side module".